mwclient/listing.py

```diff
@@ -23,9 +29,14 @@ def __init__(self, site, list_name, prefix,
         kwargs.update(args)
         self.args = kwargs
 
-        if limit is None:
-            limit = site.api_limit
-        self.args[self.prefix + 'limit'] = str(limit)
+        (max_items, api_chunk_size) = handle_limit(limit, max_items, api_chunk_size)
+
+        # for efficiency, if max_items is set and api_chunk_size is not,
+        # set the chunk size to max_items so we don't retrieve
+        # unneeded extra items (so long as it's below API limit)
+        apilimit = site.api_limit
+        api_chunk_size = api_chunk_size or min(max_items or apilimit, apilimit)
```

Comment:
nit: `api_limit` seems more Pythonic than `apilimit`

```suggestion
        api_limit = site.api_limit
        api_chunk_size = api_chunk_size or min(max_items or api_limit, api_limit)
```

Reply:
ehhh, it's just a throwaway internal variable because otherwise the line is too long :P i'd rather just call it 'limit' than 'api_limit'...